projects/app-lob/src/app/grid-dynamic-chart-data/data-analysis-dock-manager/data-analysis-dock-manager.component.ts

```diff
@@ -163,12 +194,47 @@ export class DataAnalysisDockManagerComponent implements OnInit, AfterViewInit {
         this.paneService.appendPane(splitPane);
         const chartHost = this.getChartHostFromSlot(type);
         chartHost.viewContainerRef.clear();
-        this.chartIntegration.chartFactory
         const chart = this.chartIntegration.chartFactory(type, chartHost.viewContainerRef);
 
         this.dockManager.nativeElement.layout.floatingPanes.push(splitPane);
         this.docLayout = { ...this.dockManager.nativeElement.layout };
         this.selectedCharts[type] = chart;
         this.cdr.detectChanges();
     }
-}
+
+    public createChartCommonLogic() {
+        if (Object.keys(this.selectedCharts).length !== 0) {
```

Comment:
This if is not needed.
Together with the settimeout on next line brings unnecessary complexity that is not even understandable why the code is like this.

Drop the if and document the setTimeout usage.